### PR TITLE
linter version up

### PR DESCRIPTION
**Pull request by AskAlexSharov**

### CODE REVIEW

File: erigon-lib/kv/kvcache/cache_test.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func TestEviction(t *testing.T) {
 }
 
 func TestAPI(t *testing.T) {
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS != "linux" {
```
Comment (JkLondon) [inline erigon-lib/kv/kvcache/cache_test.go:174]: why cant test on linux?

File: erigon-lib/kv/kvcache/cache_test.go [hunk 2]
```diff
@@ -171,7 +171,7 @@ func TestEviction(t *testing.T) {
...
 		t.Skip("fix me on win please")
```
Comment (JkLondon) [inline erigon-lib/kv/kvcache/cache_test.go:175]: this text should also be changed

File: polygon/heimdall/client.go [hunk 3]
```diff
@@ -140,9 +140,9 @@ func (c *Client) FetchStateSyncEvents(ctx context.Context, fromID uint64, to tim
 
 		c.logger.Trace(heimdallLogPrefix("Fetching state sync events"), "queryParams", url.RawQuery)
 
-		ctx = withRequestType(ctx, stateSyncRequest)
+		reqCtx := withRequestType(ctx, stateSyncRequest)
 
-		response, err := FetchWithRetry[StateSyncEventsResponse](ctx, c, url, c.logger)
+		response, err := FetchWithRetry[StateSyncEventsResponse](reqCtx, c, url, c.logger)
```
Comment (JkLondon) [inline polygon/heimdall/client.go:145]: why we need to save previous context?

File: erigon-lib/kv/kvcache/cache_test.go (hunk 2 above)
Comment (AskAlexSharov) [reply erigon-lib/kv/kvcache/cache_test.go:175]: it's part of another PR (which makes CI green)

**Review by JkLondon — APPROVED** [review]

**Pull request merged**